### PR TITLE
Improve a11y of form error state v2

### LINKED ISSUE #1918: Insufficient contrast in form error state
### Prerequisites

- [X] I have [searched](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

(Coming from an a11y meeting with @Aniort)

The color contrast for `:invalid` `<input>`, `<select>`, and `<textarea>` elements is not sufficient between focused and not focused states.

### Reduced test cases

We need to find a better way to convey information.

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Chrome

### What version of Boosted are you using?

v5.3.0

### PR DESCRIPTION
**Pull request by MewenLeHo**
### Related issues
 
 Closes #1918 
 
 ### Description
 
 The goal of this PR is to improve focus to allow sufficient contrast between error field focused and non focused.

**Before this PR**

Once a form is on error: fields on error will gain a red border and black one when focused
before

The main problem here is in a11y we can't convey an information by color alone because it can be not noticeable for certain types of colorblindness.
This references WCAG criterion [1.4.1 Use of Color](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html) (Level A).

For someone suffering from [achromatopsia](https://en.wikipedia.org/wiki/Achromatopsia) for instance, the same example will look like: 
before_achromatopsia

_All fields on error look almost identical, causing an a11y issue._

**After this PR**
The recent work on focus (see #1437) was applied to fields on error to add a visual cue so information does not rely exclusively on color anymore:
after

And for someone suffering from [achromatopsia](https://en.wikipedia.org/wiki/Achromatopsia):
after_achromatopsia

 ### Motivation & Context
 
 A11y improvement for people suffering for any form of colorblindness.
 
 ### Types of change
 
 - New feature (non-breaking change which adds functionality)
 
 ### Live previews
 
 - <https://deploy-preview-2090--boosted.netlify.app/docs/5.3/examples/form/>
 - https://deploy-preview-2090--boosted.netlify.app/docs/5.3/forms/validation/#supported-elements
 
 ### Checklist
 
 #### Contribution
 
 - [x] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)
 
 #### Accessibility
 
 - [x] My change follows accessibility good practices; I have at least run axe
 
 #### Design
 
 - [x] My change respects the design guidelines defined in [Orange Design System](https://system.design.orange.com/0c1af118d/p/8118d1-web)
 - [x] My change is compatible with responsive display
 
 #### Development
 
 - [x] My change follows the [developer guide](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/wiki/Developer-guide)
 - (NA) I have added JavaScript unit tests to cover my changes
 - (NA)  I have added SCSS unit tests to cover my changes
 
 #### Documentation
 
 - [x] My change introduces changes to the documentation and/or I have updated the documentation accordingly
 
 ### Checklist (for Core Team only)
 
 - [ ] My change introduces changes to the migration guide
- (NA)  My new component is added in Storybook
- (NA)  My new component is compatible with RTL
 - [x] Manually run [BrowserStack tests](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/actions/workflows/browserstack.yml)
 - [x] Manually test browser compatibility with BrowserStack (Chrome >= 60, Firefox >= 60 (+ ESR), Edge, Safari >= 12, iOS Safari, Chrome & Firefox on Android)
 - [x] Code review
 - [x] Design review
 - [x] A11y review
 
 #### After the merge
 
 - [ ] Manually launch [Percy tests](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/actions/workflows/percy.yml)

### CODE REVIEW

**Comment by MewenLeHo** [comment]
Will be a fresh start for #1949
Needed to debug old code and find a solution for at least one Firefox issue.

File: scss/_reboot.scss [hunk 1]
```diff
@@ -96,13 +96,13 @@ body {
   @include focus-visible(); // 1
 }
 
-.js-focus-visible :focus:not([data-focus-visible-added]):not(.focus-ring),
-.js-focus-visible .focus:not([data-focus-visible-added]):not(.focus-ring) { // 2
+.js-focus-visible :focus:not([data-focus-visible-added]):not(.focus-ring):not(.form-select:invalid),
```
Comment (MewenLeHo) [inline scss/_reboot.scss:99]: Seems to be needed by Chrome, but why?

File: scss/_reboot.scss [hunk 2]
```diff
@@ -96,13 +96,13 @@ body {
...
+.js-focus-visible .focus:not([data-focus-visible-added]):not(.focus-ring):not(.form-select:invalid) { // 2
   outline: 0 !important;
   box-shadow: none;
 }
 
-:focus:not(:focus-visible):not(.focus-ring) { // 3
+:focus:not(:focus-visible):not(.focus-ring):not(.form-select:invalid) { // 3
```
Comment (MewenLeHo) [inline scss/_reboot.scss:105]: Seems to be needed by Firefox, but why?

**Review by louismaximepiton — APPROVED** [review]
LGTM! To me this solution seems more logical in terms of UX, so let's wait for the design review if they are fine with it.

File: scss/_reboot.scss (hunk 1 above)
Comment (louismaximepiton) [inline scss/_reboot.scss:99]: Hmm forgot about this during the review but might need an update of the `// 2.` comment

Comment (MewenLeHo) [reply scss/_reboot.scss:101]: Just pushed a new version of the comment, could you please have a look?

**Review by louismaximepiton — APPROVED** [review]
👌

File: scss/_reboot.scss (hunk 1 above)
Comment (louismaximepiton) [reply scss/_reboot.scss:101]: Sound great, thanks!

**Comment by Franco-Riccitelli** [comment]
All looks good to me.

**Comment by julien-deramond** [comment]
Screenshot 2023-07-27 at 09 22 27

Not 100% sure but I'm wondering if adding something in the migration guide could be beneficial to end-users at least to know that something (transparent) for them when they use our components has been changed. I'm thinking about when they create their own components; they should follow the same principle. Thoughts?

**Comment by MewenLeHo** [comment]
> Form control file in https://deploy-preview-2090--boosted.netlify.app/docs/5.3/forms/validation/#supported-elements hasn't the outlines when focused with a mouse click.
> 
> Tried few things really quickly if it can help:
> 
> 1. Modify the mixin to add after the `.form-control` rule:
> 
> ```css-scss
> .form-control[type="file"] {
>       @include form-validation-state-selector($state) {
>         border-color: $border-color;
> 
>         &:focus {
>           @include focus-visible();
>           border-color: $border-color !important; // stylelint-disable-line declaration-no-important
>         }
>         // End mod
>       }
>     }
> ```
> 
> 2. Add the following 3 rules in reboot:
> 
> ```css-scss
> :not(.form-control[type="file"]:invalid)
> ```
> 
> Note: there's probably something overriding the `.form-control` rule because this use case should probably already be included and working 🤷


Yes it's really strange.
It should work without the need to add the mixin specifically for `.form-control[type="file"]` because I already added it for all `.form-control` 🤔 
I pushed the changes to make more tests.

**Comment by MewenLeHo** [comment]
@julien-deramond: as discussed in today's daily meeting, the existing rule for `.form-control` was not working for `.form-control[type="file"]` because `data-focus-visible-added` attribut is not added on click for input file 😮

File: site/content/docs/5.3/migration.md [hunk 3]
```diff
@@ -22,6 +22,10 @@ If you need more details about the changes, please refer to the [v5.3.2 release]
 - **Orange navbar**
   - <span class="badge bg-warning">Warning</span> The minimizing behavior with `.header-minimized` is applied to the `<header>` instead of one of its children directly.
 
+### Forms
+
+- <span class="badge bg-warning">Warning</span> In order to improve accessibility for people suffering for any form of colorblindness, we added a visual cue to form elements on error when focused so information does not rely exclusively on color anymore.
```
Comment (julien-deramond) [inline site/content/docs/5.3/migration.md:27]:
```suggestion
- <span class="badge bg-success">New</span> In order to improve accessibility for people suffering from any form of colorblindness, we added a visual cue to form elements on error when focused so information does not rely exclusively on color anymore. Although is has no direct impact, you might want to apply this same modification within your custom form controls if you have any.
```

**Review by julien-deramond — APPROVED** [review]
Nice! LGTM! 🚀

**Pull request merged**